pkg/checkpoint/config_map.go

```diff
@@ -12,13 +12,18 @@ import (
 
 // checkpointConfigMapVolumes ensures that all pod configMaps are checkpointed locally, then converts the configMap volume to a hostpath.
 func (c *checkpointer) checkpointConfigMapVolumes(pod *v1.Pod) (*v1.Pod, error) {
+	uid, err := podUser(pod)
+	if err != nil {
+		return nil, fmt.Errorf("failed to checkpoint configMap for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+	}
+
 	for i := range pod.Spec.Volumes {
 		v := &pod.Spec.Volumes[i]
 		if v.ConfigMap == nil {
 			continue
 		}
 
-		_, err := c.checkpointConfigMap(pod.Namespace, pod.Name, v.ConfigMap.Name)
+		_, err := c.checkpointConfigMap(pod.Namespace, pod.Name, v.ConfigMap.Name, uid, rootGID)
```

Comment:
Why use the root GID here? The `podUser` func could return the `fsGroup` from the container's securityContext if it is available, and default to `rootGID` otherwise.

Reply:
Done.